src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/MouseEventSender.cs

```diff
@@ -366,7 +366,63 @@ public static void SendMouseEvent(int replaySegment, MouseInputActionData mouseI
             }
 
             SendRawPositionMouseEvent(replaySegment, normalizedPosition, mouseInput.leftButton, mouseInput.middleButton, mouseInput.rightButton, mouseInput.forwardButton, mouseInput.backButton, mouseInput.scroll);
+        }
 
+        /// <summary>
+        /// Simulates a mouse event at the screen space position corresponding to the given Transform's world position.
+        /// </summary>
+        /// <param name="mouseTarget">The Transform whose world position will be converted to screen space for the mouse event.</param>
+        /// <param name="leftButton">Simulates left mouse button press if true.</param>
+        /// <param name="middleButton">Simulates middle mouse button press if true.</param>
+        /// <param name="rightButton">Simulates right mouse button press if true.</param>
+        /// <param name="forwardButton">Simulates forward mouse button press if true.</param>
+        /// <param name="backButton">Simulates back mouse button press if true.</param>
+        /// <param name="scroll">Simulates mouse scroll wheel movement.</param>
+        public static void SendMouseEvent(Transform mouseTarget,
+                                          bool leftButton=false,
+                                          bool middleButton=false,
+                                          bool rightButton=false,
+                                          bool forwardButton=false,
+                                          bool backButton=false,
+                                          Vector2 scroll = default)
+        {
+            Vector2 screenPosition = Camera.main.WorldToScreenPoint(mouseTarget.position);
```

Comment:
This needs a bit more processing.  The transform position of an object is seldomly the 'center' of that object.

We have code for determining that center already in `MouseEventSender.SendMouseEvent` that you can lift / use as a reference

Reply:
I spent a good amount of time but wasn't able to do this cleanly.

Initially I tried calculating the 
- bestObject.screenSpaceBounds
- whether something is the worldSpaceObject
- and possibleObjects, but it took me a long time to understand what possibleObjects are

so I switched to calling FindBestClickObject.

However, that lead me to requiring `transforms`. Which I believe I can get via objectFinders, which I can get through `var FindObjectsByType<ObjectFinder>(FindObjectsSortMode.None);` but that requires coupling this static class to a monobehavior.FindObjectsByType call and now think I am doing something wrong.

Can you give me a bit more context on what would be the best way to do this?

Reply:
@RG-nAmKcAz pinging here.

Reply:
... taking a look...

Reply:
@batu

Ultimately all that code I sent you to look at in the prior comments... uses the transforms found for the current frame.

Specifically it uses the `screenSpaceBounds` for the status of each found transform.  Said `screenSpaceBounds` are set in this code from `TransformObjectFinder`.



```csharp
var bounds = SelectBoundsForTransform(mainCamera, screenWidth, screenHeight, theTransform);
tStatus.screenSpaceBounds = bounds.Item1;
tStatus.screenSpaceZOffset = bounds.Item2;
tStatus.worldSpaceBounds = bounds.Item3;
```


So ultimately, your code should be .... 

```csharp
var boundsInfo = TransformObjectFinder.SelectBoundsForTransform(Camera.main, Screen.width, Screen.height, mouseTarget);
var ssBounds = boundsInfo.Item1;
if (ssBounds.HasValue)
{
    Vector2 screenPosition = ssBounds.Value.center;
    // do your action
}
else
{
    // error where object transform isn't on screen
}
```

Reply:
sweet thank you